AddQtAndroidApk.cmake

```diff
@@ -97,84 +98,162 @@ macro(add_qt_android_apk TARGET SOURCE_TARGET)
             set(QT_ANDROID_SDK_BUILDTOOLS_REVISION ${BUILD_TOOLS_VERSION})
         endif()
     endforeach()
-    message("Detected Android SDK build tools version ${QT_ANDROID_SDK_BUILDTOOLS_REVISION}")
 
-    # define the application source package directory
-    if(ARG_PACKAGE_SOURCES)
-        set(QT_ANDROID_APP_PACKAGE_SOURCE_ROOT ${ARG_PACKAGE_SOURCES})
+    if(ARG_VERBOSE)
+        message(STATUS "Detected Android SDK build tools version ${QT_ANDROID_SDK_BUILDTOOLS_REVISION}")
+    endif()
+
+    if(ARGS_EXTRA_QML)
+        set(QT_ANDROID_QML_IMPORT_PATH ${ARGS_EXTRA_QML})
+    endif()
+
+    set(QT_ANDROID_APP_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/${SOURCE_TARGET}-${ANDROID_ABI})
+    if(ARG_ANDROID_MANIFEST_IN_PATH)
+        set(QT_ANDROID_MANIFEST_IN_REAL_PATH ${ARG_ANDROID_MANIFEST_IN_PATH})
     else()
-        # get version code from arguments, or generate a fixed one if not provided
-        set(QT_ANDROID_APP_VERSION_CODE ${ARG_VERSION_CODE})
-        if(NOT QT_ANDROID_APP_VERSION_CODE)
-            set(QT_ANDROID_APP_VERSION_CODE 1)
-        endif()
+        set(QT_ANDROID_MANIFEST_IN_REAL_PATH ${QT_ANDROID_SOURCE_DIR}/AndroidManifest.xml.in)
+    endif()
 
+    if(ARG_VERBOSE)
+        message(STATUS "Used input AndroidManifest file QT_ANDROID_MANIFEST_IN_REAL_PATH : ${QT_ANDROID_MANIFEST_IN_REAL_PATH}")
+    endif()
+
+    # get version code from arguments, or generate a fixed one if not provided
+    set(QT_ANDROID_APP_VERSION_CODE ${ARG_VERSION_CODE})
+    if(NOT QT_ANDROID_APP_VERSION_CODE)
+        set(QT_ANDROID_APP_VERSION_CODE 1)
+    endif()
+
+    if(ARG_VERSION_NAME)
```

Comment:
We should remain consistent with the rest of the file, and not use this old CMake syntax (`if(XXX) else(XXX) endif(XXX)`) 😉

Reply:
Totally agree i only knew the old cmake way when i did that.

Reply:
Fixed in 73c857cfb89782d77744d5a4d5de10bfee0044f6